### PR TITLE
rename to opensearch and dashboard

### PR DESCRIPTION
**Pull request by prudhvigodithi**
Signed-off-by: Prudhvi Godithi <pgodithi@amazon.com>

### Description
rename to opensearch and dashboard

### Issues Resolved
https://github.com/opensearch-project/terraform-provider-opensearch/issues/18

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Hey @phillbaker @bbarani @peterzhuamazon @serge-r can you please review this rename PR, I have done some changes in few go files and refactored the code.
Thank you

**Comment by phillbaker** [comment]
I started reviewing and it mostly looks good, I'll be able to finish reviewing tomorrow.

File: provider/resource_opensearch_dashboard_object.go [hunk 1]
```diff
@@ -43,23 +43,23 @@ func resourceOpensearchKibanaObject() *schema.Resource {
 					}
 
 					for _, o := range body {
-						kibanaObject, ok := o.(map[string]interface{})
+						DashboardObject, ok := o.(map[string]interface{})
```
Comment (phillbaker) [inline provider/resource_opensearch_dashboard_object.go:46]: this shouldn't be capitalized: `dashboardObject`. Please fix all occurrences

**Review by phillbaker — APPROVED** [review]
looks good, just a couple of small thing

File: provider/resource_opensearch_dashboard_object.go (hunk 1 above)
Comment (prudhvigodithi) [reply provider/resource_opensearch_dashboard_object.go]: Thanks fixed this.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -28,29 +28,22 @@ jobs:
       matrix:
         go-version: [1.18.x]
         os: [ubuntu-latest]
-        es-major-version:
-          - "7-opendistro"
+        os-major-version:
           - "1-opensearch"
         include:
-        - es-major-version: "7-opendistro"
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply .github/workflows/test.yml:35]: Thanks I have added `7-opendistro` to be part of the matrix workflow.

**Review by rishabh6788 — APPROVED** [review]

**Pull request merged**